src/mapwindow.jl

```diff
@@ -116,6 +123,89 @@ function _mapwindow{T,N}(f,
     out
 end
 
+
+
+# This is a implementation of Fast 2D median filter
+# http://ieeexplore.ieee.org/document/1163188/
+function median_filter{T,N}(f,
+                         img::AbstractArray{T,N},
+                         window::Indices{N},
+                         border::BorderSpecAny,
+                         shape=default_shape(f);
+                         callmode=:copy!)
+    inds = indices(img)
```

Comment:
I have created new dispatches for different methods and separated implementations for discrete and continuous inputs. Do suggest, if there can be a better dispatch method.
Presently: 
1. Two new functions are added 
   a. median_direct!
   b. median_histogram!
2. Only for img::Union{AbstractArray{N0f8,N},AbstractArray{ColorTypes.Gray{N0f8},N}} is a choice made between median_histogram! and median_direct!.